assets/javascripts/discourse/initializers/decrypt-posts.js

```diff
@@ -435,9 +429,7 @@ export default {
               });
           })
           .catch(() => {
-            state.encryptState = "error";
-            state.error = I18n.t("encrypt.invalid_identity");
-            this.scheduleRerender();
+            showModal("activate-encrypt", { model: this });
```

Comment:
Can you clarify what `this` is here? It seems to be the context of `decryptPost` which is a function? A model should probably be an object instance of some sort.

Reply:
`decryptPost` is called in the context of a widget (see [this](https://github.com/discourse/discourse-encrypt/blob/8b995f8acae77213c1b01d3353cf355fffeb8086/assets/javascripts/discourse/initializers/decrypt-posts.js#L330)), so `this` refers to a `post-contents` widget. I changed `model` data structure with something more descriptive.